### PR TITLE
feat: disable user management changes if in community edition [DET-8791]

### PR DESCRIPTION
**Pull request by billboggs**
## Description

Ticket: [DET-8791](https://determinedai.atlassian.net/browse/DET-8791)

Disabling user updates in the community edition as the user changes will be handled elsewhere. 

## Test Plan

- Go to user settings pages in deployments outside of the community edition. Validate user update actions are still enabled.
- Go to user settings pages in community edition deployments. Validate user update actions are not enabled. 

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-8791]: https://determinedai.atlassian.net/browse/DET-8791?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/api_user.go [hunk 1]
```diff
@@ -237,6 +238,9 @@ func (a *apiServer) GetUserByUsername(
 func (a *apiServer) PostUser(
 	ctx context.Context, req *apiv1.PostUserRequest,
 ) (*apiv1.PostUserResponse, error) {
+	if len(a.m.config.InternalConfig.ExternalSessions.LoginURI) > 1 {
```
Comment (ioga) [inline master/internal/api_user.go:241]: could we have a `func (c* Config) IsCommunityEdition(...` in `internal/config/config.go` as a helper for this somewhat non-obvious condition?

**Comment by ClaireNeveu** [comment]
We try to maintain a one-codebase approach for the frontend code to simplify maintenance. It would be better to just make this UI work correctly in OSS.

**Comment by mackrorysd** [comment]
I think IsCommunityEdition() is the wrong name - it's going to cause confusion between different products (as it has with Claire - because this is not about an OSS / MLDE distinction). What we're really switching on is being part of a managed offering where user management is entirely external, is NOT done via SCIM / SAML, and that doesn't have a final name yet. Can we just call it "ExternalSessions", but still refactor the conditional we've been using into a clearer function?

**Comment by billboggs** [comment]
Updated naming to `ExternalSessions.Enabled` and added the `userManagementEnabled` config.

**Review by trentwatt — APPROVED** [review]
LGTM, great work!

**Comment by trentwatt** [comment]
Circle CI had an outage, but hopefully the tests will run tonight, if not you may have to rerun:

From 20 mins ago 
image

**Pull request merged**